flaky test: TestResumeFromCheckpointE2EWithManualSentinel

```
2026-02-25T19:43:34.3615307Z test-1           | --- FAIL: TestResumeFromCheckpointE2EWithManualSentinel (49.90s)
2026-02-25T19:43:34.3616032Z test-1           |     runner_resume_test.go:1064: 
2026-02-25T19:43:34.3616667Z test-1           |         	Error Trace:	/app/pkg/migration/runner_resume_test.go:1064
2026-02-25T19:43:34.3617525Z test-1           |         	Error:      	Condition never satisfied
2026-02-25T19:43:34.3618222Z test-1           |         	Test:       	TestResumeFromCheckpointE2EWithManualSentinel
2026-02-25T19:43:34.3619055Z test-1           |         	Messages:   	migration did not reach WaitingOnSentinelTable
```

--- a/pkg/migration/check_constraint_test.go
+++ b/pkg/migration/check_constraint_test.go
@@ -135,7 +135,8 @@ func TestCheckConstraintWithDML(t *testing.T) {
 
 	m := NewTestRunner(t, "chk_dml", "ENGINE=InnoDB",
 		WithThreads(1),
-		WithTargetChunkTime(100*time.Millisecond))
+		WithTargetChunkTime(100*time.Millisecond),
+		WithTestThrottler())
 
 	var wg sync.WaitGroup
 	wg.Go(func() {

--- a/pkg/migration/datatype_test.go
+++ b/pkg/migration/datatype_test.go
@@ -213,7 +213,8 @@ func testEnumReorder(t *testing.T, enableBuffered bool) {
 	m := NewTestRunner(t, "enumreorder", "MODIFY COLUMN status ENUM('pending', 'active', 'inactive') NOT NULL",
 		WithThreads(1),
 		WithTargetChunkTime(100*time.Millisecond),
-		WithBuffered(enableBuffered))
+		WithBuffered(enableBuffered),
+		WithTestThrottler())
 
 	// Concurrent DML during copy phase to exercise binlog replay.
 	ctx, cancel := context.WithCancel(t.Context())
@@ -301,7 +302,8 @@ func testSetReorder(t *testing.T, enableBuffered bool) {
 	m := NewTestRunner(t, "setreorder", "MODIFY COLUMN perms SET('execute', 'read', 'write') NOT NULL",
 		WithThreads(1),
 		WithTargetChunkTime(100*time.Millisecond),
-		WithBuffered(enableBuffered))
+		WithBuffered(enableBuffered),
+		WithTestThrottler())
 
 	// Concurrent DML during copy phase.
 	ctx, cancel := context.WithCancel(t.Context())
@@ -524,7 +526,8 @@ func testAlterPKIntToBigIntWithDML(t *testing.T, enableBuffered bool) {
 	)`)
 	tt.SeedRows(t, "INSERT INTO altpk_dml (name, val) SELECT 'seed', 1", 4096)
 	m := NewTestRunner(t, "altpk_dml", "MODIFY COLUMN id BIGINT NOT NULL AUTO_INCREMENT",
-		WithBuffered(enableBuffered))
+		WithBuffered(enableBuffered),
+		WithTestThrottler())
 
 	var wg sync.WaitGroup
 	wg.Go(func() {
@@ -611,7 +614,8 @@ func testAlterPKIntToBigIntWithDMLAndAdditionalColumnChange(t *testing.T, enable
 		"MODIFY COLUMN id BIGINT NOT NULL AUTO_INCREMENT, MODIFY COLUMN name VARCHAR(255) NOT NULL",
 		WithThreads(2),
 		WithTargetChunkTime(100*time.Millisecond),
-		WithBuffered(enableBuffered))
+		WithBuffered(enableBuffered),
+		WithTestThrottler())
 
 	var wg sync.WaitGroup
 	wg.Go(func() {

--- a/pkg/migration/resume_test.go
+++ b/pkg/migration/resume_test.go
@@ -859,6 +859,7 @@ func TestResumeFromCheckpointE2EWithManualSentinel(t *testing.T) {
 		WithDBName(dbName),
 		WithThreads(1),
 		WithTargetChunkTime(100*time.Millisecond),
+		WithTestThrottler(),
 		WithRespectSentinel())
 
 	ctx, cancel := context.WithCancel(t.Context())

